src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
> If I understood correctly, we don't need to concern about the cases like that since it is not compilable.

Yes.

> This seems weird we treat break and return with the same variable as there is a difference between them and how they are handled.

@Luolc Here is two examples I have created that isn't requested to be `final` with your code.

````
    void foo1() throws Exception {
        Exception e; // violation
        final int a = (int) Math.random();
        final int b = (int) Math.random();

        while (b == 0) {
            if (a == 0) {
                e = new Exception();
                return;
            }
        }

        e = new Exception();
        throw e;
    }

    void foo2() throws Exception {
        Exception e; // violation
        final int a = (int) Math.random();
        final int b = (int) Math.random();

        switch (a) {
        case 0:
            e = new Exception();
            return;
        default:
            break;
        }

        e = new Exception();

        throw e;
    }
````

No violation is correct if `return` was treated like `break`, but `return` and `break` mean different things.
Regression doesn't show, this as I mentioned, because it only reports violations (added or removed).